src/System.Text.Json/tests/Utf8JsonWriterTests.cs

```diff
@@ -1313,36 +1674,116 @@ public void WriteComments(bool formatted, bool skipValidation, string comment)
+        private static void WriteCommentValue(Utf8JsonWriter jsonUtf8, int i, string comment)
         {
             switch (i)
             {
                 case 0:
-                    jsonUtf8.WriteCommentValue(comment, escape: false);
+                    jsonUtf8.WriteCommentValue(comment);
                     break;
                 case 1:
-                    jsonUtf8.WriteCommentValue(comment.AsSpan(), escape: false);
+                    jsonUtf8.WriteCommentValue(comment.AsSpan());
                     break;
                 case 2:
-                    jsonUtf8.WriteCommentValue(Encoding.UTF8.GetBytes(comment), escape: false);
+                    jsonUtf8.WriteCommentValue(Encoding.UTF8.GetBytes(comment));
                     break;
             }
         }
 
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public void WriteInvalidComment(bool formatted, bool skipValidation)
+        {
+            var options = new JsonWriterOptions { Indented = formatted, SkipValidation = skipValidation };
+
+            var output = new ArrayBufferWriter<byte>(32);
+            var jsonUtf8 = new Utf8JsonWriter(output, options);
+
+            string comment = "comment is */ invalid";
+
+            Assert.Throws<ArgumentException>(() => jsonUtf8.WriteCommentValue(comment));
+            Assert.Throws<ArgumentException>(() => jsonUtf8.WriteCommentValue(comment.AsSpan()));
+            Assert.Throws<ArgumentException>(() => jsonUtf8.WriteCommentValue(Encoding.UTF8.GetBytes(comment)));
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public void WriteCommentsInvalidTextAllowed(bool formatted, bool skipValidation)
+        {
+            var options = new JsonWriterOptions { Indented = formatted, SkipValidation = skipValidation };
+
+            var output = new ArrayBufferWriter<byte>(32);
+            var jsonUtf8 = new Utf8JsonWriter(output, options);
+
+            string comment = "comment is * / valid";
+            jsonUtf8.WriteCommentValue(comment);
+            jsonUtf8.WriteCommentValue(comment.AsSpan());
+            jsonUtf8.WriteCommentValue(Encoding.UTF8.GetBytes(comment));
+
+            comment = "comment is /* valid";
+            jsonUtf8.WriteCommentValue(comment);
+            jsonUtf8.WriteCommentValue(comment.AsSpan());
+            jsonUtf8.WriteCommentValue(Encoding.UTF8.GetBytes(comment));
+
+            comment = "comment is / * valid even with unpaired surrogate \udc00 this part no longer visible";
+            jsonUtf8.WriteCommentValue(comment);
+            jsonUtf8.WriteCommentValue(comment.AsSpan());
+
+            jsonUtf8.Flush();
+
+            // Explicitly skipping flushing here
+            var invalidUtf8 = new byte[2] { 0xc3, 0x28 };
```

Comment:
This test captures the current semantics for writing comments that contain invalid data. Let me know if its fine as is.